templates/typescript/samples/sample-skill/src/dialogs/mainDialog.ts

```diff
@@ -130,18 +125,18 @@ export class MainDialog extends ComponentDialog {
             // Check for any interruptions
             const interrupted = await this.interruptDialog(innerDc);
 
-            if (interrupted) {
-                // If dialog was interrupted, return EndOfTurn
-                return MainDialog.EndOfTurn;
+            if (interrupted !== undefined) {
+                // If dialog was interrupted, return interrupted result
+                return interrupted;
             }
         }
 
         return await super.onContinueDialog(innerDc);
     }
 
     // Runs on every turn of the conversation to check if the conversation should be interrupted.
-    protected async interruptDialog(innerDc: DialogContext): Promise<Boolean> {
-        let interrupted = false;
+    protected async interruptDialog(innerDc: DialogContext): Promise<DialogTurnResult> {
+        let interrupted: DialogTurnResult = {} as DialogTurnResult;
```

Comment:
```suggestion
    protected async interruptDialog(innerDc: DialogContext): Promise<DialogTurnResult> | undefined {
        let interrupted: DialogTurnResult | undefined = undefined;
```

Taking into account that [C#](https://github.com/microsoft/botframework-solutions/blob/v1.0/samples/csharp/skill/SkillSample/Dialogs/MainDialog.cs#L118) can return `null` and when it's called to this method always [checks](https://github.com/southworks/botframework-solutions/pull/384/files#diff-031ce6a60277e736381a0f80207d3607R128) if it's undefined or not, I think that we can do this.

What do you think?

Reply:
If this doesn't make a snowball effect and the expected parameter needs to be changed to `DialogTurnResult | undefined` in subsequent methods consuming this, I'm ok

Reply:
I guess not, as it's correctly validated when it's used. You can double check it!

Reply:
I'm getting this error when applying this
`TS2322: Type 'DialogTurnResult<any> | undefined' is not assignable to type 'DialogTurnResult<any>'.   Type 'undefined' is not assignable to type 'DialogTurnResult<any>'.`
Perhaps we should find another solution

Reply:
Ack, leave it as you did. It's ok!